primitives/src/lib.rs

```diff
@@ -149,10 +149,25 @@ impl TradingPair {
 		self.1
 	}
 
+	pub fn swap(&self) -> TradingPair {
+		TradingPair(self.second(), self.first())
+	}
+
 	pub fn dex_share_currency_id(&self) -> CurrencyId {
 		CurrencyId::join_dex_share_currency_id(self.first(), self.second())
 			.expect("shouldn't be invalid! guaranteed by construction")
 	}
+
+	pub fn from_currency_ids_unordered(currency_id_a: CurrencyId, currency_id_b: CurrencyId) -> Option<Self> {
+		if (currency_id_a.is_token_currency_id() || currency_id_a.is_erc20_currency_id())
+			&& (currency_id_b.is_token_currency_id() || currency_id_b.is_erc20_currency_id())
+			&& currency_id_a != currency_id_b
+		{
+			Some(TradingPair(currency_id_a, currency_id_b))
+		} else {
+			None
+		}
+	}
```

Comment:
Hmmm, current structure of trading paths in dex-aggregator, use (DOT, AUSD) (AUSD, BTC) as a trading path rather than [DOT, AUSD, BTC]. Perhaps that is not possible due to TradingPair structure

Reply:
We want to ensure TradingPair is ordered so we won't have DOT/aUSD pair and aUSD/DOT pair at same time.

You can hover make a new `struct TradingDirection { from: CurrencyId, to: CurrencyId }` or something similar

Reply:
Yeah, that's what I'm thinking too. I will refactor so TradingPair logic is not touched